packages/react/src/evm/hooks/async/contracts.ts

```diff
@@ -149,7 +152,11 @@ export type UseContractResult<
  */
 export function useContract<
   TContract extends ValidContractInstance = SmartContract,
->(contractAddress: RequiredParam<ContractAddress>) {
+  TContractType extends ContractType = "custom",
```

Comment:
wait so I can pass 2 conflicting types here?
like `useContract<NFTDrop, "marketplace">(...)` ?

Reply:
yes, but the runtime type would take precedence (because it actually changes behavior)